src/proc/pointcloud.cpp

```diff
@@ -101,11 +104,87 @@ namespace librealsense
         set_extrinsics();
     }
 
+    template< class callback >
+    calibration_change_callback_ptr create_calibration_change_callback_ptr( callback&& cb )
+    {
+        return {
+            new rs2::calibration_change_callback< callback >( std::move( cb ) ),
+            []( rs2_calibration_change_callback* p ) { p->release(); }
+        };
+    }
+
     void pointcloud::inspect_other_frame(const rs2::frame& other)
     {
         if (_stream_filter != _prev_stream_filter)
         {
             _prev_stream_filter = _stream_filter;
+
+            if (!_registered_auto_calib_cb)
+            {
+                auto sensor = ((frame_interface*)other.get())->get_sensor();
+                if (sensor)
+                {
+                    auto dev = sensor->get_device().shared_from_this();
+                    device_calibration * d2r = dynamic_cast<device_calibration*>(dev.get());
+                    assert( d2r );
```

Comment:
The assert is L515-only. Should be either if/els-ed or removed

Reply:
Done

Reply:
👍